autoload/ale/events.vim

```diff
@@ -138,6 +138,32 @@ function! ale#events#Init() abort
                 " not be changed here.
                 autocmd InsertLeave * if exists('*ale#engine#Cleanup') | call ale#cursor#EchoCursorWarning() | endif
             endif
+
+            function! AleShowCursorDetailCmd() abort
+                if exists('*ale#engine#Cleanup')
+                    call ale#preview#CloseIfTypeMatches('ale-preview')
+                    " Don't jump to the preview window during cursor movement.
+                    call ale#cursor#ShowCursorDetail({'stay_here': 1})
+                endif
+            endfunction
+
+            if g:ale_show_cursor_detail
+                if len(g:ale_cursor_detail_fileextensions)
+                    let ftStr=""
+
+                    for ft in g:ale_cursor_detail_fileextensions
+                        let ftStr .= ft . ","
+                    endfor
+
+                    execute "autocmd CursorMoved,CursorHold *.{" . strpart(ftStr, 0, len(ftStr)-1) . "} call AleShowCursorDetailCmd()"
+                else
+                    autocmd CursorMoved,CursorHold * call AleShowCursorDetailCmd()
+                endif
+            endif
+
+            if g:ale_close_preview_on_insert
+                autocmd InsertEnter * call ale#preview#CloseIfTypeMatches('ale-preview')
```

Comment:
Wrap the command here in `if exists('*ale#preview#CloseIfTypeMatches(') | ... | endif`. That way we won't load the `preview.vim` file when you enter Insert mode, without having yet shown a preview window.